python/hsfs/training_dataset.py

```diff
@@ -108,8 +114,13 @@ def save(self, features, write_options={}):
             )
 
         user_version = self._version
+        user_stats_config = self._statistics_config
         self._features = engine.get_instance().parse_schema(feature_dataframe)
         self._training_dataset_engine.save(self, feature_dataframe, write_options)
+        # currently we do not save the training dataset statistics config for training datasets
+        self.statistics_config = user_stats_config
```

Comment:
Why are you doing this? (assigning to a different variable?)

Reply:
for training dataset we aren't saving the config in the database along the metadata. the `engine.save()` before will update `self` with the metadata from the backend, hence overwriting the config. I am saving it in this temporary variable `user_stats_config` so I can set it again afterwards

Reply:
ok